numba/tests/test_struct_ref.py

```diff
@@ -226,6 +229,14 @@ def test_MyStructType_in_dict_mixed_type_error(self):
             # the second field is now an integer.
             td['b'] = MyStruct(2.3, 1)
 
+    def test_MyStructType_hash_no_typeof_recursion(self):
+        '''Tests that __hash__ is not called prematurely in typeof 
+           causing infinite recursion (see #8241).'''
+        st = MyStruct(1, 2)
+        typeof(st)
+
+        assert hash(st) == 3
```

Comment:
```suggestion
        self.assertEqual(hash(st), 3)
```
Numba uses `unittest` style assertions.

Reply:
Ahh getting too used to pytest.